File: backend/src/main/java/net/archasmiel/skufapi/dto/ErrorResponse.java
package net.archasmiel.skufapi.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Response for error")
public class ErrorResponse implements IResponse {

  @Schema(description = "Error", example = "GOOGLE_AUTH_FAILED")
  private String error;

  @Schema(description = "Message", example = "This error means...")
  private String message;
}

File: backend/src/main/kotlin/net/archasmiel/skufapi/api/response/ErrorResponse.kt
package net.archasmiel.skufapi.api.response

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Response for error")
data class ErrorResponse(
    @field:Schema(description = "Error code", example = "GOOGLE_AUTH_FAILED")
    val code: String,

    @field:Schema(description = "Message, representing user data", example = "Auth failed due to...")
    val message: String
) : ApiResponse